.dockerignore

```diff
@@ -1,2 +0,0 @@
-*
-!docker/
```

Comment:
This is definitely a bad thing, as for people using docker to build projects it means copying everything to the docker daemon, git repos and maybe even GBs of test data (it happened in the past). Isn't there any way to specify a different `.dockerignore` file?

Reply:
Main blocker here is makd. It is trivial to change `.dockerignore` so that it ignores all `.git` folders (including submodule ones) which makes context size small enough to not worry about it. But makd requires git repository context to work and there doesn't seem to be any way to provide missing pieces of information manually.

Reply:
Ah, right. Still, basically anything that is not part of `git ls-files` or `.git` should be ignored.

Reply:
I think I was using whitelist originally (with just things like `src`, `pkg` and `doc` being copied) but reverted that because of makd. We don't need tests or examples to generate package either.

Reply:
Yeah, but source code is light. I'm still thinking of regular users using docker, where dirty repos will be the norm, and as I said it happened to me in the past that docker took a long time because I had some several GB test files. It took me some time to figure it out.

Reply:
Whitelist solves that. I think makd is really the only obstacle here right now.